### PR TITLE
Add syscall for BLS aggregate signatures

### PR DESCRIPTION
**Pull request by DrPeterVanNostrand**
This PR reopens the previously closed PR #1836.

### CODE REVIEW

File: fvm/src/kernel/default.rs [hunk 1]
```diff
@@ -511,14 +511,26 @@ where
...
             .iter()
-            .map(|len| {
-                let (plaintext, rem) = plaintexts_concat.split_at(*len as usize);
-                plaintexts_concat = rem;
-                plaintext
+            .map(|&len| {
+                let start = offset;
+                offset = start
+                    .checked_add(len as usize)
+                    .context("invalid bls signature length")
```
Comment (DrPeterVanNostrand) [inline fvm/src/kernel/default.rs:521]: Should this be `invalid bls plaintext length`?

File: fvm/src/gas/price_list.rs [hunk 2]
```diff
@@ -102,19 +101,14 @@ lazy_static! {
...
-                    scale: Gas::new(26),
-                },
-            }
-        },
         secp256k1_recover_cost: Gas::new(1637292),
+        bls_pairing_cost: Gas::new(8299302),
+        bls_hashing_cost: ScalingCost {
+            flat: Gas::zero(),
+            // TODO: Can we just use the "sha256" number below?
+            scale: Gas::new(7),
```
Comment (DrPeterVanNostrand) [inline fvm/src/gas/price_list.rs:109]: Just a note: this PR did not change the BLS signature hash function (that maps a plaintext's bytes to a curve point). I'm guessing that this gas price change (to that of SHA256's) was intentional as SHA256 is used internally by the BLS hash function.

Comment (Stebalien) [reply fvm/src/gas/price_list.rs:109]: Yeah, that makes sense. I'll leave it as its own separate price just to be safe.

File: fvm/src/kernel/default.rs (hunk 1 above)
Comment (Stebalien) [reply fvm/src/kernel/default.rs]: Yep!

**Comment by Stebalien** [comment]
Merged in #2003

**Pull request closed without merging**